src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/VisibleMemberTable.java

```diff
@@ -141,23 +141,44 @@ public static Set<Kind> forDetailsOf(ElementKind kind) {
         }
     }
 
+    /** The class or interface described by this table. */
     private final TypeElement te;
+    /**
+     * The superclass of {@link #te} or null if {@code te} is an
+     * interface or {@code java.lang.Object}.
+     */
     private final TypeElement parent;
 
     private final BaseConfiguration config;
     private final BaseOptions options;
     private final Utils utils;
     private final VisibleMemberCache mcache;
 
+    /** Tables for direct and indirect superclasses. */
     private final List<VisibleMemberTable> allSuperclasses;
+    /**
+     * Tables for direct and indirect superinterfaces.
+     * Tables for superinterfaces might not be unique.
+     */
     private final List<VisibleMemberTable> allSuperinterfaces;
+    /**
+     * Tables for direct superclass and direct superinterfaces.
+     *
+     * The position of a table for the superclass in the list is unspecified.
+     * Tables for the superinterfaces might not be unique (i.e. an interface
+     * may be added from different lineages).
+     */
     private final List<VisibleMemberTable> parents;
```

Comment:
Should this be something different, such as a Set or a multi-map?
Generally, javac and to some extent javadoc have historically used lists for "list-y sets".